WordPress/gutenberg
# Legacy widget's preview functionality is broken when the page is moved

## Description
Legacy Widget block has a preview function. However, it uses a relative URL.
`iframe` cannot load the widgets.php file when the code is no longer running from the /wp-admin/ URL.
This PR fixes that by using a REST API endpoint instead of the widgets.php file to fetch preview HTML. It doesn't depend on the admin URL.
Fixes #30049 

Based on #34230

## How has this been tested?
1. Go to Appearance -> Widgets page.
2. Click on the "plus sign" button at the very bottom of the page.
3. Add a `Legacy Widget` block.
4. Select `Meta` widget from the dropdown.
5. Click somewhere on the background of the page (not on the `Meta` widget itself). You should see a preview of the `Meta` widget now.
6. Open your browser's developer tools and find an `<iframe />` element with the `wp-block-legacy-widget__edit-preview-iframe`  class. Check its `srcdoc` attribute.

#### Expected result:
The `srcdoc` attribute of the `<iframe />` should contain html code of the preview.
Please see the screenshots section for reference.

## Screenshots 
![Screenshot 2021-08-12 at 20 00 39](https://user-images.githubusercontent.com/43744263/129246203-e3734aba-d1fc-4ec1-b8e8-6d381e3d488d.png)
![Screenshot 2021-08-19 at 21 59 05](https://user-images.githubusercontent.com/43744263/130135924-2d876802-ac20-4b01-94ca-de47ddf48756.png)


## Types of changes
Bug fix.

## Checklist:
